### DISCUSSION TITLE
Windows11でコンパイルするとビルドエラーが発生する

### DISCUSSION THREAD

**Discussion by ohfujimasa73** [nuta/microkernel-book#4] [Help]
### 概要

makeを実施後に下記ビルドエラーが発生する

エラー内容
>error: unable to create target: 'No available targets are compatible with triple "riscv32"'
>1 error generated.
>make: *** [build/kernel/main.o] エラー 1
※Ubuntuは実施可能

clang　バージョンは下記
PS C:\Users\user\Desktop\HinaOS\microkernel-book> clang --version
compnerd.org clang version 14.0.0
Target: x86_64-unknown-windows-msvc
Thread model: posix

**Comment by nuta** [comment]
@ohfujimasa73 さん

ご報告ありがとうございます。Windowsアプリケーション専用の構成でビルドされたclangが参照されているようです。おそらく、Visual Studioか何かのものが紛れ込んでいるのでしょう。PowerShellを開いて、以下のように `gcm clang` の実行結果を共有していただけますでしょうか。

```
PS C:\> gcm clang
 
CommandType     Name                Version    Source
-----------     ----                -------    ------
Application     clang.exe           0.0.0.0    C:\Program Files\llvm\bin\clang.exe
```

**Comment by ohfujimasa73** [comment]
@nuta さん

>PowerShellを開いて、以下のように gcm clang の実行結果を共有していただけますでしょうか。

下記結果になります

CommandType     Name      　Version    Source
-----------     　　----             -------    ------
Application     　clang.exe     0.0.0.0    C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin\clang.exe

上記未使用のため削除しましたらビルド通りました。ありがとうございます。

**Reply by nuta** [reply]
もしかしてSwiftが入っていますか？手元の環境で色々試してみたところ、次のようにSwiftが別の`clang`をインストールするようです。

```
PS C:\Users\seiya> gcm -All clang

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     clang.exe                                          0.0.0.0    C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin\clang.exe
Application     clang.exe                                          0.0.0.0    C:\Program Files\llvm\bin\clang.exe
```

このclangはWindowsアプリケーション専用ビルドらしく、HinaOSのビルド (エラーが指し示す通り`riscv32`用のコンパイル) ができないようになっています。

[HinaOSのビルドシステムの方での修正](https://github.com/nuta/microkernel-book/pull/6) を先ほどマージしたので、最新のリポジトリの内容をプルしていただければ修正が取り込まれます。

なお、消したファイルはSwiftのツールチェーンを再インストールすると復活するはずです。 😃
